[bitnami/mariadb] fix missing volumeMounts key for metrics container

tstollin opened the pull request:
Signed-off-by: Stollin, Thomas <thomas.stollin@interhyp.de>

<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing! We will try to test and integrate the change as soon as we can, but be aware we have many GitHub repositories to manage and can't immediately respond to every request. There is no need to bump or check in on a pull request (it will clutter the discussion of the request).

 Also don't be worried if the request is closed or not integrated sometimes the priorities of Bitnami might not match the priorities of the pull request. Don't fret, the open source community thrives on forks and GitHub makes it easy to keep your changes in a forked repo.
 -->

### Description of the change

<!-- Describe the scope of your change - i.e. what the change does. -->

This change fixes a templating error caused by adding extraVolumeMounts to the mariadb metrics container when customPasswordFiles are used. This combination of configurations results in a missing ```volumeMounts``` key for the metrics container and dangling list entries. 
The problem was introduced with https://github.com/bitnami/charts/pull/11142. 

### Benefits

<!-- What benefits will be realized by the code change? -->

### Possible drawbacks

<!-- Describe any known limitations with your change -->

### Applicable issues

<!-- Enter any applicable Issues here (You can reference an issue using #) -->
  - fixes #

### Additional information

<!-- If there's anything else that's important and relevant to your pull request, mention that information here.-->

### Checklist

<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

migruiz4 commented on bitnami/mariadb/templates/primary/statefulset.yaml:
```diff
@@ -318,14 +318,16 @@ spec:
           resources: {{- toYaml .Values.metrics.resources | nindent 12 }}
           {{- end }}
-          {{- if and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles) }}
+          {{- if or (and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles)) .Values.metrics.extraVolumeMounts.primary }}
           volumeMounts:
+          {{- if and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles) }}
             - name: mariadb-credentials
               mountPath: /opt/bitnami/mysqld-exporter/secrets/
           {{- end }}
           {{- if .Values.metrics.extraVolumeMounts.primary }}
           {{- include "common.tplvalues.render" (dict "value" .Values.metrics.extraVolumeMounts.primary "context" $) | nindent 12 }}
           {{- end }}
+          {{- end }}
```
The extra conditionals are not really needed, could you please refactor this as it follows?

```suggestion
          volumeMounts:
          {{- if and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles) }}
            - name: mariadb-credentials
              mountPath: /opt/bitnami/mysqld-exporter/secrets/
          {{- end }}
          {{- if .Values.metrics.extraVolumeMounts.primary }}
          {{- include "common.tplvalues.render" (dict "value" .Values.metrics.extraVolumeMounts.primary "context" $) | nindent 12 }}
          {{- end }}
```

tstollin replied:
I added the conditionals as I was not sure how Kubernetes would react, if the key `volumeMounts` is rendered, but without any list entries (not even an empty list). This would happen if neither `.Values.auth.usePasswordFiles` nor `.Values.metrics.extraVolumeMounts` are used. But I trust your expertise and I will follow your suggestion.

migruiz4 replied:
Hi @tstollin, no worries. 
Kubernetes won't fail if an empty `volumeMounts` is rendered, the same approach is used in other charts, e.g. airflow:
https://github.com/bitnami/charts/blob/21cc5f9ab8127da23f271f085a2e8bd916c2cc0f/bitnami/airflow/templates/worker/statefulset.yaml#L182-L202

tstollin replied:
Perfect. Thanks for the clarification.

migruiz4 commented on bitnami/mariadb/Chart.yaml:
```diff
@@ -26,4 +26,4 @@ sources:
   - https://github.com/prometheus/mysqld_exporter
   - https://mariadb.org
-version: 11.1.1
+version: 11.1.2
```
Version 11.1.2 has already been released, could you please update this to version `11.1.3`?

migruiz4 reviewed (changes requested):
Hi @tstollin,

Thank you very much for your contribution. Could you please take a look at my comments below?

tstollin commented:
@migruiz4 I implemented your suggestions. While doing this, I noticed that the same bug existed in the statefulset.yaml of the secondary containers. I fixed this yaml as well. 
Please have a look.

tstollin commented:
The CI/CD Pipeline blocks a merge because of findings by trivy. As far as I understand the logs, there are security findings in the mariadb docker image. As the image is not part of my changes, I'm unsure what I can do to pass the checks. :(
Any help is appreciated!

carrodher commented:
In order to fix the failing action, could you please rebase from master? Thanks

migruiz4 reviewed (approved):
Thank you very much for your contribution @tstollin!
